IonDotnet/Internals/SharedSymbolTable.cs

```diff
@@ -110,7 +110,8 @@ public SymbolToken Find(string text)
                 return default;
 
             var internedText = _symbolNames[sid - 1];
-            return new SymbolToken(internedText, sid);
+            // Normalize the SID and import reference to this table.
+            return new SymbolToken(internedText, sid, new ImportLocation(Name, sid));
```

Comment:
Can we refactor this constructor so we do not need to pass "sid" twice? (once as the second param and once within the ImportLocation object)